annotation-processor/src/main/java/org/hibernate/validator/ap/checks/MethodParameterOverrideCheck.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.ap.checks;
+
+import org.hibernate.validator.ap.util.CollectionHelper;
+
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Checks, that the given parameter has a correct set of constraints if it's overriding a method form other class.
+ * Applies to methods parameters.
+ */
+public class MethodParameterOverrideCheck extends AbstractConstraintCheck {
+
+	private static final String JAVA_LANG_OBJECT = "java.lang.Object";
+
+	private Elements elementUtils;
+
+	private Types typeUtils;
+
+	public MethodParameterOverrideCheck(Elements elementUtils, Types typeUtils) {
+		this.elementUtils = elementUtils;
+		this.typeUtils = typeUtils;
+	}
+
+	@Override
+	public Set<ConstraintCheckError> checkField(VariableElement element, AnnotationMirror annotation) {
+		// get a current class in which a method is located.
```

Comment:
I'm confused that this is implemented in `checkField()`. Shouldn't it rather be `checkMethod()`?

Thinking of it, do you support validation of parameter/return value constraints on constructors, too (I guess that's more for your other PR on constraint sanity, though).